Sources/Basic/ShellEscapedString.swift

```diff
@@ -0,0 +1,52 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+private let whitelist: Set<Character> = {
+    Set("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_/:@#%+=.,".characters.map{$0})
+}()
+
+public extension String {
+
+    /// Creates a shell escaped string. If the string does not need escaping, returns the original string.
+    /// Otherwise escapes using single quotes. For eg: hello -> hello, hello$world -> 'hello$world', input A -> 'input A'
+    ///
+    /// - Returns: Shell escaped string.
+    public func shellEscaped() -> String {
+        // If all the characters in the string are in whitelist then no need to escape.
+        guard let pos = characters.index(where: { !whitelist.contains($0) }) else {
```

Comment:
I suspect it is significantly faster to iterate over the `utf8`, and maybe to not use a set but manually encode the set into a switch statement. Would be nice to have a performance test of this, even if we end up disabling it in CI (using `#if false`).